src/CarbonSDK.ts

```diff
@@ -340,6 +351,22 @@ class CarbonSDK {
     return this.connect(wallet);
   }
 
+  public async connectWithLeap(
+    leap: Leap,
+    opts?: CarbonWalletGenericOpts,
+  ) {
+    const chainId = await LeapAccount.getChainId(this);
```

Comment:
why are we passing in the whole SDK into the `getChainId` function. we should initialise the a `chain_id` field in the SDK and use that instead.